src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -373,14 +373,18 @@ static int add_partition_swap(DissectedPartition *p) {
         if (!arg_swap_enabled)
                 return 0;
 
-        /* Disable the swap auto logic if at least one swap is defined in /etc/fstab, see #6192. */
-        r = fstab_has_fstype("swap");
-        if (r < 0)
-                return log_error_errno(r, "Failed to parse fstab: %m");
-        if (r > 0) {
-                log_debug("swap specified in fstab, ignoring.");
-                return 0;
+        /* Prioritize fstab for honoring custom swap options, see #6192. */
+        r = fstab_has_node(p->node);
+        if (r < 0) {
```

Comment:
Maybe `if (r == -ENOTRECOVERABLE)` is better?

Reply:
Hmm, I figure we should be conservative here, i.e. always fall back on any sort of error

Reply:
Then, the`-ENOTRECOVERABLE` commit is mostly meaningless??

Reply:
No. Previously we'd return false if the device nodes are missing...

Reply:
'Previously' means without this commit?? I cannot follow??

Reply:
fstab_is_mount_point_full() needs to ignore ENOENT for the paths listed in fstab, with the exact same reason that we can't check the presence of that path in fstab_is_same_node. The first commit makes sure if udev is really not around we return something workable error rather than assuming the entry didn't exist in fstab.